documentation/docs/developers/connectors/task4.md

```diff
@@ -1,8 +1,178 @@
+
+```python
+RATE_LIMITS = [
+    # Pools
+    RateLimit(limit_id=REQUEST_WEIGHT, limit=1200, time_interval=ONE_MINUTE),
+    RateLimit(limit_id=ORDERS, limit=10, time_interval=ONE_SECOND),
+    RateLimit(limit_id=ORDERS_24HR, limit=100000, time_interval=ONE_DAY),
+    # Weighted Limits
+    RateLimit(limit_id=SNAPSHOT_PATH_URL, limit=MAX_REQUEST, time_interval=ONE_MINUTE,
+              linked_limits=[LinkedLimitWeightPair(REQUEST_WEIGHT, 50)]),
+    RateLimit(limit_id=BINANCE_CREATE_ORDER, limit=MAX_REQUEST, time_interval=ONE_MINUTE,
+              linked_limits=[LinkedLimitWeightPair(REQUEST_WEIGHT, 1),
+                             LinkedLimitWeightPair(ORDERS, 1),
+                             LinkedLimitWeightPair(ORDERS_24HR, 1)]),
+]
+```
+
+!!! note
+    Binance implements API Pools as well as weighted requests. In the example above, the `BINANCE_CREATE_ORDER` endpoint has a request weight of 1 for 3 API Pools, while the `SNAPSHOT_PATH_URL` endpopint has a request weight of 50 for the `REQUEST_WEIGHT` API Pool. Notice that the API Pools have different rate limits and time intervals.
+
+## Integrating Rate Limits into the connector
+
+The throttler should be consumed by all relevant classes that issues an API endpoints. Namely the `Exchange/Derivative`, `APIOrderBookDataSource` and `UserStreamDataSource` classes. Doing so ensures that the throttler manages all REST API/Websocket requests issued by any of the connector components.
+
+In this example, we will use referencing the `NdaxExchange` class.
+
+### Initializing the Connector
+
+In the ```__init__()``` function, we have to initialize the `AsyncThrottler`.
+
+```python
+from hummingbot.connector.exchange.ndax import ndax_constants as CONSTANTS
+
+class NdaxExchange(ExchangeBase):
+  def __init__(...):
+    ...
+    self._throttler = AsyncThrottler(CONSTANTS.RATE_LIMITS)
+    self._order_book_tracker = NdaxOrderBookTracker(
+            throttler=self._throttler, trading_pairs=trading_pairs, domain=domain
+        )
+        self._user_stream_tracker = NdaxUserStreamTracker(
+            throttler=self._throttler, auth_assistant=self._auth, domain=domain
+        )
+```
+
+!!! note
+    Notice that we pass the throttler as arguments to the `NdaxOrderBookTracker` as well as `NdaxUserStreamTracker`. This is to ensure that all classes that interact with the Ndax API servers share the same throttler.
+
+### Consuming the throttler
+
+To consume the throttler when executing an API request, we simple wrap the contents of the `_api_request(...)` function within the an asynchrounous context as seen below.
+
+```python
+def _api_request(...):
+  async with self._throttler.execute_task(path_url):
+    ...
+```
+
+!!! warning
+    The `path_url` must be match the `limit_id` of the endpoint as defined in the `RATE_LIMITS` constant. The throttler will match the `path_url` to its assigned rate limits or API pools.
```

Comment:
I think that, when first introducing the rate limit IDs above, it's worth explaining that the limit ID's can actually be entirely arbitrary, but it helps to assign the endpoint's name as ID to help with debugging, for example

Reply:
```suggestion
    The `path_url` must match the `limit_id` of the endpoint as defined in the `RATE_LIMITS` constant. The throttler will match the `path_url` to its assigned rate limits or API pools.
```